style-guide.md

```diff
@@ -0,0 +1,8 @@
+# Style Guide
+
+We strongly recommend to follow [Google's Developer Documentation Style Guide](https://developers.google.com/style) when writing developer docs except the following:
+
+- Title cases are used in capitalization in titles and headings, rather than [sentence case](https://developers.google.com/style/capitalization#capitalization-in-titles-and-headings). Use [this website](https://titlecase.com/) to convert to proper title cases.
+- When referencing to Subnet in text, upper case S as in **S**ubnet is always used.
+- Official project/product names:
+  - Subnet-EVM
```

Comment:
Can we add the rest of our products, `Avalanche-CLI`, `AvalancheGo`, `Subnet-CLI`, `CoreEth`, any others? That's the correctly spelling for `Avalanche-CLI`. I'm not sure if the others are right.